pkg/template/event_test.go

```diff
@@ -57,8 +62,10 @@ func Test_BodyPathVarRegex_invalid(t *testing.T) {
 		"$[body]",
 		"${body}",
 		"$(body.)",
-		"$(body..)",
```

Comment:
Why was this test case removed? I think `$(body..)` would be invalid, right?

Reply:
Yes indeed, however if we want to support having a `.` in the key of a dict then we cannot easily avoid that, unless we split the regex to say what the first character may or may not be and then every other character from the second onward. I though it would not be worth it.
I added more used cases though that used to be considered valid before and are now not valid anymore.

Reply:
I don't think `gjson` works with `.` as a key of a dict, because it messes up the syntax; here's a go playground link: https://play.golang.org/p/OTQkYyFSJaM. Let me know if I messed up the syntax or am missing something here.

Reply:
The "." is a valid character as long as it's escaped.
See the syntax reference https://github.com/tidwall/gjson/blob/master/SYNTAX.md#escape-character

The "." is important to filter labels, that often are namespaced and the namespace includes a ".", for instance:

```
spec:
  params:
  - name: pipelineRun
    value: $(body.Data.taskRun.metadata.labels.tekton\.dev/pipelineRun)
```

Reply:
So, that might be a  bug in the gjson library. I don't think it works when I try to escape the `.` character: https://play.golang.org/p/8f67D3_JTrU

Reply:
heh, you need to escape the back-slash for go, else go thinks you're trying to define a string an escape sequence "\." rather than a string with the "\" character followed by the "." one. 
https://play.golang.org/p/16L6fO6LHMT

Reply:
Ohhhhh, lol, thanks! 😅